src/configuration/vimrc.ts

```diff
@@ -0,0 +1,96 @@
+import * as _ from 'lodash';
+import * as fs from 'fs';
+import * as path from 'path';
+import { IConfiguration, IKeyRemapping, IVimrcKeyRemapping } from './iconfiguration';
+import { vimrcKeyRemappingBuilder } from './vimrcKeyRemappingBuilder';
+
+class VimrcImpl {
+  public load(configuration: IConfiguration) {
+    if (configuration.vimrcPath) {
+      configuration.vimrcPath = VimrcImpl.expandHome(configuration.vimrcPath);
+      if (!fs.existsSync(configuration.vimrcPath)) {
+        return;
+      }
+    } else {
+      configuration.vimrcPath = VimrcImpl.findDefaultVimrc();
+      if (!configuration.vimrcPath) {
+        return;
+      }
+    }
+
+    let vimrcContent = fs.readFileSync(configuration.vimrcPath, { encoding: 'utf8' });
+    let lines = vimrcContent.split(/\r?\n/);
+
+    for (const line of lines) {
+      VimrcImpl.tryKeyRemapping(configuration, line);
+    }
+  }
+
+  private static tryKeyRemapping(configuration: IConfiguration, line: string): boolean {
+    let mapping: IVimrcKeyRemapping | null = vimrcKeyRemappingBuilder.build(line);
+    if (!mapping) {
+      return false;
+    }
+
+    let collection: IKeyRemapping[];
+    switch (mapping.keyRemappingType) {
+      case 'nmap':
+        collection = configuration.normalModeKeyBindings;
+        break;
+      case 'vmap':
+        collection = configuration.visualModeKeyBindings;
+        break;
+      case 'imap':
+        collection = configuration.insertModeKeyBindings;
+        break;
+      case 'nnoremap':
+        collection = configuration.normalModeKeyBindingsNonRecursive;
+        break;
+      case 'vnoremap':
+        collection = configuration.visualModeKeyBindingsNonRecursive;
+        break;
+      case 'inoremap':
+        collection = configuration.insertModeKeyBindingsNonRecursive;
+        break;
+      default:
+        return false;
+    }
+
+    // Don't override a mapping present in settings.json; those are more specific to VSCodeVim.
+    if (!collection.some(r => _.isEqual(r.before, mapping!.keyRemapping.before))) {
+      collection.push(mapping.keyRemapping);
```

Comment:
What if duplicates are added? What if there is a malformed remapping -- can we have these settings undergo `src/configuration/validators/remappingValidator.ts`?

Reply:
If things go wrong, how do users debug this?

Reply:
Actually, I don't think it's possible that a bad remapping could be generated, based on what's currently being validated in `remappingValidator`. That said, remappings from .vimrc are added to the config before the config is validated.